### PR TITLE
Update express

### PR DESCRIPTION
**Pull request by raychan**

### CODE REVIEW

File: definitions/npm/express_v4.x.x/flow_v0.32.x-/express_v4.x.x.js [hunk 1]
```diff
@@ -100,7 +97,6 @@ declare class express$Response extends http$ServerResponse mixins express$Reques
   type(type: string): this;
   vary(field: string): this;
   req: express$Request;
-  end: (data: string | Buffer, encoding: string) => this;
```
Comment (raychan) [inline definitions/npm/express_v4.x.x/flow_v0.32.x-/express_v4.x.x.js]: @briancavalier From express's document, end comes from Node core, specifically the [response.end() method of http.ServerResponse.](https://nodejs.org/api/http.html#http_response_end_data_encoding_callback) And the type definition is [here](https://github.com/facebook/flow/blob/master/lib/node.js#L1290) and it is different from our definition. But I can't figure out how to make [this part ](https://github.com/nowait/yelp-gil-service/blob/master/src/requestHandler/logging.js#L43-L57) type check. Any thoughts?

**Comment by raychan** [comment]
Ok, @briancavalier I think this is ready for review.
I also created a [PR](https://github.com/nowait/yelp-gil-service/pull/241) on yelp_gil for this

**Comment by raychan** [comment]
I will see if I can add some tests

File: definitions/npm/express_v4.x.x/flow_v0.32.x-/express_v4.x.x.js [hunk 2]
```diff
@@ -40,6 +44,7 @@ declare class express$Request extends http$IncomingMessage mixins express$Reques
   header(field: string): string | void;
   is(type: string): boolean;
   param(name: string, defaultValue?: string): string | void;
+  accepts(types: Array<string>): string | false;
```
Comment (briancavalier) [inline definitions/npm/express_v4.x.x/flow_v0.32.x-/express_v4.x.x.js]: :+1: One suggestion: move it up to line 41, so it's near the other `accepts` variant.

File: definitions/npm/express_v4.x.x/flow_v0.32.x-/express_v4.x.x.js [hunk 3]
```diff
@@ -89,6 +94,7 @@ declare class express$Response extends http$ServerResponse mixins express$Reques
   status(statusCode: number): this;
   type(type: string): this;
   vary(field: string): this;
+  req: express$Request;
```
Comment (briancavalier) [inline definitions/npm/express_v4.x.x/flow_v0.32.x-/express_v4.x.x.js:97]: It'll be interesting to see if we get feedback on this one when we PR it back to flow-types.  Express definitely uses it internally, but I wonder if it's something they don't want to expose publicly.  Let's keep it and see what happens 😄

**Review by briancavalier — CHANGES_REQUESTED** [review]
Looks good.  Only one minor suggestion

**Comment by raychan** [comment]
@briancavalier addressed your feedback and add few tests

File: definitions/npm/express_v4.x.x/flow_v0.32.x-/test_response.js [hunk 4]
```diff
@@ -14,6 +14,7 @@ app.use('/response_api', (req: express$Request, res: express$Response) => {
     'Content-Type': 'application/json',
   });
   res.end('{}');
+  console.log('request: ', res.req);
```
Comment (briancavalier) [inline definitions/npm/express_v4.x.x/flow_v0.32.x-/test_response.js]: Leftover console.log?

File: definitions/npm/express_v4.x.x/test_express_v4.x.x.js [hunk 5]
```diff
@@ -133,6 +133,7 @@ app.use('/something', (req: express$Request, res: express$Response) => {
 app.use((err: ?Error, req, res, next) => {
     // test req
     req.accepts('accepted/type');
+    req.accepts(['json', 'text']);
```
Comment (briancavalier) [inline definitions/npm/express_v4.x.x/test_express_v4.x.x.js:136]: :+1:

**Review by briancavalier — CHANGES_REQUESTED** [review]
I think there's a leftover console.log.  Other than that, lgtm

File: definitions/npm/express_v4.x.x/flow_v0.32.x-/test_response.js (hunk 4 above)
Comment (raychan) [reply definitions/npm/express_v4.x.x/flow_v0.32.x-/test_response.js]: Haha, this is actually a test. 😛  It just pop up in my head first. Any other suggestion?

**Comment by raychan** [comment]
I opened a issue on flow's repo to improve http method's type definition.
https://github.com/facebook/flow/issues/4500

File: definitions/npm/express_v4.x.x/flow_v0.32.x-/test_response.js (hunk 4 above)
Comment (raychan) [reply definitions/npm/express_v4.x.x/flow_v0.32.x-/test_response.js]: I updated this test to match other [similar tests](https://github.com/nowait/flow-typed/blob/master/definitions/npm/express_v4.x.x/flow_v0.32.x-/test_overrideUseMethodClassExtension.js#L55)

File: definitions/npm/express_v4.x.x/flow_v0.32.x-/test_response.js [hunk 6]
```diff
@@ -14,7 +14,7 @@ app.use('/response_api', (req: express$Request, res: express$Response) => {
     'Content-Type': 'application/json',
   });
   res.end('{}');
-  console.log('request: ', res.req);
+  res.req;
```
Comment (briancavalier) [inline definitions/npm/express_v4.x.x/flow_v0.32.x-/test_response.js]: Wow.  This works?  Does it actually verify the type?  I was going to suggest:

```js
const r: $Request = res.req
```

But yours is even simpler 😄 (if it's verifying the type!)

**Comment by briancavalier** [comment]
Looks good.  One question [here](https://github.com/nowait/flow-typed/pull/2/files/fca9c5d40145bd51c97dcf51f82923f497f9f8b9..34287e847cd37c0b41f8dbfd682dac4ca9401f61#r130766585) about the `res.req` test.  Otherwise, LGTM

File: definitions/npm/express_v4.x.x/flow_v0.32.x-/test_response.js (hunk 6 above)
Comment (raychan) [reply definitions/npm/express_v4.x.x/flow_v0.32.x-/test_response.js]: yes, here are some similar tests: https://github.com/nowait/flow-typed/blob/master/definitions/npm/express_v4.x.x/flow_v0.32.x-/test_overrideUseMethodClassExtension.js#L54-L59

**Pull request merged**